modules/ingest-ecs/src/test/java/org/elasticsearch/ingest/ecs/EcsNamespaceProcessorTests.java

```diff
@@ -0,0 +1,409 @@
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_scopeNotMap() {
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", "not a map");
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_invalidAttributes() {
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        source.put("attributes", "not a map");
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_invalidBody() {
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        source.put("body", "not a map");
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_invalidBodyText() {
+        Map<String, Object> body = new HashMap<>();
+        body.put("text", 123);
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        source.put("body", body);
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_invalidBodyStructured() {
+        Map<String, Object> body = new HashMap<>();
+        body.put("structured", "a string");
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        source.put("body", body);
+        assertFalse(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testIsOTelDocument_validBody() {
+        Map<String, Object> body = new HashMap<>();
+        body.put("text", "a string");
+        body.put("structured", new HashMap<>());
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        source.put("body", body);
+        assertTrue(EcsNamespaceProcessor.isOTelDocument(source));
+    }
+
+    public void testExecute_validOTelDocument() {
+        Map<String, Object> source = new HashMap<>();
+        source.put("resource", new HashMap<>());
+        source.put("scope", new HashMap<>());
+        Map<String, Object> body = new HashMap<>();
+        body.put("text", "a string");
+        body.put("structured", new HashMap<>());
+        source.put("body", body);
+        source.put("key1", "value1");
+        Map<String, Object> before = new HashMap<>(source);
+        IngestDocument document = new IngestDocument("index", "id", 1, null, null, source);
+        processor.execute(document);
+        assertEquals(before, document.getSource());
```

Comment:
I don't think this test is actually testing what its intended to test -- `new HashMap<>(source)` should probably be some kind of deep copy instead.

Reply:
Actually, nevermind, I rewrote it in 65f7ea2a2f613b2a5e3d62d69f9c12880e88cf77.

Reply:
It tested what _I intended_ it to test - that keys are not moved from the top level when you process a valid OTel document. A shallow copy is sufficient for this purpose.
In fact, now it doesn't test anything - the source always remains the same instance, you can add this assertion to all tests and they will pass.
I will restore that, keeping your nicer map construction.

Reply:
> In fact, now it doesn't test anything 

I don't think I agree -- since it's an immutable map of immutable maps, the fact that `processor.execute(document)` doesn't throw an exception is probably a suitable proxy for it not having attempted to modify the map.

Reply:
Eh, I'm not going to fight about it. I like it as I wrote it here, but I'll probably be okay with some different version of things, too.